### PR TITLE
[1068] Remove non continuous applications logic from `back_links.rb`

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

Continuous applications was built using a feature flag. Now we’ve launched it we’re left with a lot of redundant code relating to the “old flow”. We are removing that now. 

This PR deals with `back_links.rb`.

## Guidance to review

I have amended some of the tests to work for continuous applications and removed some which are now tested in `spec/system/candidate_interface/continuous_applications/carry_over/candidate_carries_over_spec.rb` 

There was an edge case where a test was failing because it was testing an old reference flow.

I recommend reviewing commit by commit. 

## Link to Trello card

https://trello.com/c/CiW9H8RG/1068-catd-concerns-back-links

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/controllers/concerns/back_links.rb [hunk 1]
```diff
@@ -21,34 +21,20 @@ def redirect_back_to_application_review_page?
...
-      old_cycle_application_form_path
-    end
-  end
-  module_function :application_form_path
-
-  def continuous_application_form_path
     if request.path.match?(/withdraw/)
       candidate_interface_continuous_applications_choices_path
     else
       candidate_interface_continuous_applications_details_path
```
Comment (avinhurry) [inline app/controllers/concerns/back_links.rb:33]: Thoughts on renaming `candidate_interface_continuous_applications_choices_path` to `candidate_interface_choices_path`. It's in 42 files.

and `candidate_interface_continuous_applications_details_path` to `candidate_interface_details_path`. It's in 28 files.

**Comment by avinhurry** [comment]
[CATD: Concerns – back links](https://trello.com/c/CiW9H8RG/1068-catd-concerns-back-links)

File: spec/system/provider_interface/provider_views_application_submitted_in_new_cycle_feature_flag_disabled_spec.rb [hunk 2]
```diff
@@ -1,185 +0,0 @@
...
-    and_i_am_permitted_to_see_the_candidates_application
-    and_i_sign_in_to_the_provider_interface
-
-    when_i_visit_the_provider_page
-    then_i_can_see_a_filter_for_the_current_recruitment_cycle_year
-    then_i_can_see_and_load_the_candidates_application
-  end
-
-  def given_the_feature_flag_is_disabled
-    FeatureFlag.deactivate(:one_personal_statement)
```
Comment (avinhurry) [inline spec/system/provider_interface/provider_views_application_submitted_in_new_cycle_feature_flag_disabled_spec.rb:48]: Created a card to look at these old feature flags: https://trello.com/c/eus6FksO/1084-tech-debt-clean-up-feature-flags

File: app/controllers/concerns/back_links.rb (hunk 1 above)
Comment (JR-G) [reply app/controllers/concerns/back_links.rb:33]: We should definitely do it, but I'm happy for that to go in a separate card, since it has a wider impact than just the back links

File: spec/system/candidate_interface/entering_details/candidate_entering_other_qualification_spec.rb [hunk 3]
```diff
@@ -375,12 +372,4 @@ def and_i_delete_my_remaining_qualifications
     when_i_click_on_delete_my_first_qualification
     and_i_confirm_that_i_want_to_delete_my_additional_qualification
   end
-
-  def when_i_click_back_to_application_form
-    click_link 'Back to application'
```
Comment (vassyz) [inline spec/system/candidate_interface/entering_details/candidate_entering_other_qualification_spec.rb:380]: You can still keep these tests, but check for the details page. Something like:
```ruby
def when_i_click_back_to_my_details
  click_link 'Back to your details'
end

def then_i_see_the_your_details_page
  within '.govuk-heading-xl' do
    expect(page).to have_content 'Your details'
  end
end
```

Comment (avinhurry) [reply spec/system/candidate_interface/entering_details/candidate_entering_other_qualification_spec.rb:380]: Cheers, have updated 👍🏼

**Review by vassyz — APPROVED** [review]
Excellent!
